server/queries/MapQueries.cpp

```diff
@@ -111,12 +111,12 @@ bool CGarrisonDialogQuery::blocksPack(const CPack * pack) const
 
 	if(auto arts = dynamic_ptr_cast<ExchangeArtifacts>(pack))
 	{
-		if(auto id1 = std::visit(GetEngagedHeroIds(), arts->src.artHolder))
-			if(!vstd::contains(ourIds, *id1))
+		if(auto id1 = arts->src.artHolder)
```

Comment:
I think GetEngagedHeroIds struct is no longer used and can be removed?

Reply:
sure